common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/MqttTransportHandler.java

```diff
@@ -928,6 +929,9 @@ public void doDisconnect() {
                 gatewaySessionHandler.onGatewayDisconnect();
             }
             deviceSessionCtx.setDisconnected();
+            context.onAuthFailure(address);
```

Comment:
Looks like you should add these three lines to the "onDeviceDeleted" method instead of calling this every time the "doDisconnect" is executed.

Reply:
Moved